pkg/oc/cli/admin/release/extract.go

```diff
@@ -66,7 +90,39 @@ type ExtractOptions struct {
 	ImageMetadataCallback func(m *extract.Mapping, dgst digest.Digest, config *docker10.DockerImageConfig)
 }
 
-func (o *ExtractOptions) Complete(cmd *cobra.Command, args []string) error {
+func (o *ExtractOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	switch {
+	case len(args) == 0 && len(o.From) == 0:
+		cfg, err := f.ToRESTConfig()
+		if err != nil {
+			return fmt.Errorf("info expects one argument, or a connection to a 4.0 OpenShift server: %v", err)
+		}
+		client, err := configv1client.NewForConfig(cfg)
+		if err != nil {
+			return fmt.Errorf("info expects one argument, or a connection to a 4.0 OpenShift server: %v", err)
+		}
+		cv, err := client.Config().ClusterVersions().Get("version", metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return fmt.Errorf("you must be connected to a v4.0 OpenShift server to fetch the current version")
+			}
+			return fmt.Errorf("info expects one argument, or a connection to a 4.0 OpenShift server: %v", err)
+		}
+		image := cv.Status.Desired.Image
+		if len(image) == 0 && cv.Spec.DesiredUpdate != nil {
+			image = cv.Spec.DesiredUpdate.Image
```

Comment:
Maybe worth adding a warning we're looking into new version and not the current one.

Reply:
That's not a warning.  The desired state is what the cluster shows.

Reply:
Also, you can have a desired state without a valid status (although it would be rare)